app/main.py

```diff
@@ -1,34 +1,90 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple[int, int], end: tuple[int, int]) -> None:
+        self.decks = self._create_decks(start, end)
+        self.is_drowned = False
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def _create_decks(self, start: tuple[int, int],
+                      end: tuple[int, int]) -> list[Deck]:
+        decks = []
+        start_row, start_col = start
+        end_row, end_col = end
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+        if start_row == end_row:
+            for col in range(start_col, end_col + 1):
+                decks.append(Deck(start_row, col))
+        elif start_col == end_col:
+            for row in range(start_row, end_row + 1):
+                decks.append(Deck(row, start_col))
+        return decks
+
+    def get_deck(self, row: int, column: int) -> Deck | None:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+        return None
+
+    def fire(self, row: int, column: int) -> bool:
+        deck: Deck | None = self.get_deck(row, column)
+        if deck and deck.is_alive:
+            deck.is_alive = False
+            if all(not d.is_alive for d in self.decks):
+                self.is_drowned = True
+            return True
+        return False
 
```

Comment:
The `fire` method in the `Ship` class has a redundant `return` statement. According to the checklist, you should avoid using `return` statements that don't do anything. In this case, the `return` statement on line 40 is not needed because the function will return `None` by default if it doesn't hit any of the `return` statements in the `if` conditions. Here's how you can rewrite this method:

python
    def fire(self, row: int, column: int) -> bool:
        deck: Deck | None = self.get_deck(row, column)
        if deck and deck.is_alive:
            deck.is_alive = False
            if all(not d.is_alive for d in self.decks):
                self.is_drowned = True
            return True
```
This version of the method is cleaner and more Pythonic.